Tools/Scripts/run-mvt-tests

```diff
@@ -0,0 +1,305 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2024 Igalia S.L.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin St, Fifth Floor,
+# Boston, MA 02110-1301, USA.
+# pylint: disable=missing-docstring,invalid-name
+
+"""MVT WebDriver test runner"""
+
+import os
+import sys
+
+scripts_dir = os.path.dirname(os.path.abspath(__file__))
+if os.path.isdir(os.path.join(scripts_dir, 'webkitpy')):
+    sys.path.insert(0, scripts_dir)
+    import webkitpy
+
+import argparse
+import json
+import subprocess
+import time
+
+import subprocess
+from subprocess import CalledProcessError
+
+from selenium import webdriver
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.wpewebkit.options import Options
+
+top_level_directory = os.path.normpath(os.path.join(os.path.dirname(__file__), '..', '..'))
+
+TEST_SUITES = [
+    "codec-support-test",
+    "dash-html5-test",
+    "dash-shaka-test",
+    "dash-dashjs-test",
+    "hls-shaka-test",
+    "hls-hlsjs-test",
+    "hss-html5-test",
+    "hss-dashjs-test",
+    "progressive-html5-test",
+]
+
+COG_PLATFORMS = [
+    "drm",
+    "fdo",
+    "gtk4",
+    "headless",
+    "wl",
+]
+
+DEFAULT_PORT = "8088"
+DEFAULT_BROWSER = "/usr/bin/cog"
+DEFAULT_CONFIGURATION = "release"
+DEFAULT_PLATFORM = "headless"
+DEFAULT_TEST_SUITE = "dash-shaka-test"
+
+def fatal(*args):
+    print()
+    print("Fatal:", end=' ')
+    print(*args)
+    sys.exit(1)
+
+def parse_args():
+    """Parse command line arguments"""
+
+    parser = argparse.ArgumentParser(
+        description="Run MVT suite with WebDriver.",
+        epilog="""
+            This script connects to a running WPEWebDriver and fires up a MVT
+            test suite, collecting the results in a JSON file.
+
+            It also expects a running MVT instance in the environment
+            variable $MVT_INSTANCE_ADDRESS.
+
+            For more info, check https://github.com/rdkcentral/mvt
+            """,
+    )
+
+    parser.add_argument("address", help="Address of the remote WebDriver (i.e 0.0.0.0:8088)")
```

Comment:
Instead of having the user to guess a port number here can we just automate it?
1. Pick one random port number between 1024 and 10000
2. Check that the port is free: https://stackoverflow.com/a/52872579
3. if is not free then repeat from 1

Reply:
In case the user doesn't set a port, the port 8088 is used by default:

https://github.com/WebKit/WebKit/pull/28091/files#diff-a86141502a4c4312c94880365b24458f8541a0c945bfe37e41fec15edc4d5fbcR103

Perhaps can make more clear that the port number is not mandatory.